beacon-chain/db/kv/blocks.go

```diff
@@ -729,10 +798,26 @@ func (s *Store) slotRootsInRange(ctx context.Context, start, end primitives.Slot
 
 	var pairs []slotRoot
 	key := bytesutil.SlotToBytesBigEndian(end)
+
+	edge := false // used to detect whether we are at the very beginning or end of the index
 	err := s.db.View(func(tx *bolt.Tx) error {
 		bkt := tx.Bucket(blockSlotIndicesBucket)
 		c := bkt.Cursor()
-		for k, v := c.Seek(key); k != nil; k, v = c.Prev() {
+		for k, v := c.Seek(key); ; /* rely on internal checks to exit */ k, v = c.Prev() {
+			if len(k) == 0 && len(v) == 0 {
+				// The `edge`` variable and this `if` deal with 2 edge cases:
+				// - Seeking past the end of the bucket (the `end` param is higher than the highest slot).
+				// - Seeking before the beginning of the bucket (the `start` param is lower than the lowest slot).
+				// In both of these cases k,v will be nil and we can handle the same way using `edge` to
+				// - continue to the next iteration. If the following Prev() key/value is nil, Prev has gone past the beginning.
+				// - Otherwise, iterate as usual.
```

Comment:
Do we have a unit test for these particular edge cases ?

Reply:
Yes, `TestStore_Blocks_FiltersCorrectly` that touch the upper bounds of the blocks with `SetStartSlot` + `SetEndSlot`. LMK if that is sufficient, or if you think we need a separate test.